client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -401,6 +400,23 @@ function mediaButton( editor ) {
 
 		// Determine the next usable size
 		let sizeIndex = SIZE_ORDER.indexOf( parsed.appearance.size );
+
+		// Get the closest possible size when the current size is unknown
+		if ( sizeIndex === -1 ) {
```

Comment:
This looks like it's trying to achieve a similar goal as the `do while` block below. Could we simply update the `while` condition there to allow the loop to be run even when `sizeIndex` is `-1`?